File: benchmarks/pandas/bench_dataframe_to_string.py
"""Benchmark: DataFrame.to_string on 1k-row pandas DataFrame"""
import json, time
import pandas as pd

ROWS = 1_000
WARMUP = 3
ITERATIONS = 10
df = pd.DataFrame({"a": range(ROWS), "b": [i * 1.5 for i in range(ROWS)]})

for _ in range(WARMUP):
    df.to_string()

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.to_string()
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "dataframe_to_string", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_dataframe_to_string.ts
/**
 * Benchmark: dataFrameToString on 1k-row DataFrame
 */
import { DataFrame, dataFrameToString } from "../../src/index.js";

const ROWS = 1_000;
const WARMUP = 3;
const ITERATIONS = 10;
const a = Array.from({ length: ROWS }, (_, i) => i);
const b = Array.from({ length: ROWS }, (_, i) => i * 1.5);
const df = DataFrame.fromColumns({ a, b });

for (let i = 0; i < WARMUP; i++) dataFrameToString(df);
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) dataFrameToString(df);
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "dataframe_to_string",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
